src/state/actions/userActions.js

```diff
@@ -30,8 +30,29 @@ export const signUp = createAsyncThunk('user/signup', async user => {
   }
 });
 
+export const editProfile = createAsyncThunk('user/editProfile', async payload => {
+  try {
+    if (payload.email && payload.currentPassword) {
+      const { data } = await userService.editProfile(payload);
+      await userService.changePassword(payload);
```

Comment:
Here you are loosing `changePassword` response data right?
You could either store it in a separate `data` variable or even use `Promise.all()` to make both calls in parallel and automatically have both responses in an array

Reply:
Yes that's right. I initially opted for this approach because I thought the data returned from changePassword request was not necessary, but I realized it actually is. I changed it to `Promise.all()` but there's a bit of a problem when any of the promises provided rejects, I didn't find a way to return the response payload to the corresponding reducer, so error messages are not showing up correctly.